substrate-archive/src/archive.rs

```diff
@@ -32,129 +32,147 @@ use sp_runtime::{
-	pub block_workers: Option<usize>,
+	block_workers: usize,
 	/// Number of 64KB Heap pages to allocate for wasm execution
-	pub wasm_pages: Option<u64>,
-	/// Chain spec describing the chain
-	pub chain_spec: Option<Box<dyn ChainSpec>>,
-	pub _marker: PhantomData<(B, R, D, DB)>,
+	wasm_pages: u64,
 	/// maximum amount of blocks to index at once
-	pub max_block_load: Option<u32>,
+	max_block_load: u32,
 }
 
-impl<B, R, D, DB> Default for Builder<B, R, D, DB> {
+impl<B, R, D, DB> Default for ArchiveBuilder<B, R, D, DB> {
 	fn default() -> Self {
+		let num_cpus = num_cpus::get();
 		Self {
+			_marker: PhantomData,
+			chain_spec: None,
 			chain_data_path: None,
-			cache_size: None,
 			pg_url: None,
-			block_workers: None,
-			wasm_pages: None,
-			chain_spec: None,
-			_marker: PhantomData,
-			max_block_load: None,
+			cache_size: 128,                  // 128 MB
+			block_workers: num_cpus,          // the number of logical cpus in the system
+			wasm_pages: 64 * num_cpus as u64, // 64 * (number of logic cpu's)
+			max_block_load: 100_000,          // 100_000 blocks to index at once
 		}
 	}
 }
 
-impl<B, R, D, DB> Builder<B, R, D, DB> {
+impl<B, R, D, DB> ArchiveBuilder<B, R, D, DB> {
+	/// Specify a chain spec for storing metadata about the running archiver
+	/// in a persistant directory.
+	///
+	/// # Default
+	/// Defaults to storing metadata in a temporary directory.
+	pub fn chain_spec(mut self, spec: Box<dyn ChainSpec>) -> Self {
+		self.chain_spec = Some(spec);
+		self
+	}
+
 	/// Set the chain data backend path to use for this instance.
 	///
 	/// # Default
 	/// defaults to the environment variable CHAIN_DATA_DB
-	pub fn chain_data_db<S: Into<String>>(mut self, path: S) -> Self {
-		self.chain_data_path = Some(path.into());
+	pub fn chain_data_path<S: Into<PathBuf>>(mut self, path: Option<S>) -> Self {
+		self.chain_data_path = path.map(Into::into);
 		self
 	}
 
 	/// Set the url to the Postgres Database
 	///
 	/// # Default
 	/// defaults to value of the environment variable DATABASE_URL
-	pub fn pg_url<S: Into<String>>(mut self, url: S) -> Self {
-		self.pg_url = Some(url.into());
+	pub fn pg_url<S: Into<String>>(mut self, url: Option<S>) -> Self {
+		self.pg_url = url.map(Into::into);
 		self
 	}
 
 	/// Set the amount of cache Rocksdb should keep.
 	///
 	/// # Default
 	/// defaults to 128MB
-	pub fn cache_size(mut self, cache_size: usize) -> Self {
-		self.cache_size = Some(cache_size);
+	pub fn cache_size(mut self, cache_size: Option<usize>) -> Self {
+		if let Some(cache_size) = cache_size {
+			self.cache_size = cache_size;
+		}
 		self
 	}
 
 	/// Set the number of threads spawn for block execution.
 	///
 	/// # Default
 	/// defaults to the number of logical cpus in the system
-	pub fn block_workers(mut self, workers: usize) -> Self {
-		self.block_workers = Some(workers);
+	pub fn block_workers(mut self, workers: Option<usize>) -> Self {
+		if let Some(block_workers) = workers {
```

Comment:
Why only configure it if it (is_some) was already set?

This needs to be documented if that's the intention